include/dds/sub/DataReader.hpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2020, Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OMG_DDS_SUB_DATAREADER_HPP_
+#define OMG_DDS_SUB_DATAREADER_HPP_
+
+#include <dds/sub/detail/DataReader.hpp>
+
+#include <dds/core/Entity.hpp>
+#include <dds/sub/qos/DataReaderQos.hpp>
+#include <dds/sub/Subscriber.hpp>
+
+namespace dds {
+namespace sub {
+
+class DataReaderListener;
+
+/**
+ * @brief
+ * A DataReader allows the application to declare the data it wishes to receive.
+ *
+ * @see @ref DCPS_Modules_Datareader "Datareader"
+ */
+class DataReader : public dds::core::TEntity<detail::DataReader>
```

Comment:
I'm not sure about deleting the template, maybe it is better to leave the PSM API for later